src/service/resources-impl.js

```diff
@@ -1113,32 +1115,41 @@ export class Resources {
   /**
    * Schedules change of the element's height.
    * @param {!Resource} resource
-   * @param {number} newHeight
+   * @param {number|undefined} newHeight
+   * @param {number|undefined} newWidth
    * @param {boolean} force
    * @param {function=} opt_callback A callback function.
    * @private
    */
-  scheduleChangeHeight_(resource, newHeight, force, opt_callback) {
-    resource.resetPendingChangeHeight();
-    if (resource.getLayoutBox().height == newHeight) {
+  scheduleChangeSize_(resource, newHeight, newWidth, force, opt_callback) {
+    resource.resetPendingChangeSize();
+    const layoutBox = resource.getLayoutBox();
+    const isHeightOnly = newHeight && !newWidth;
```

Comment:
Issue with `0` - check against `undefined`. You can always try something like this:

```
if (
     (newHeight === undefined || newHeight == layoutBox.height) &&
     (newWidth === undefined || newWidth == layoutBox.width)
   ) {
  return;
}
```